server/src/main/java/org/opensearch/search/aggregations/bucket/histogram/DateHistogramAggregator.java

```diff
@@ -117,34 +117,44 @@ class DateHistogramAggregator extends BucketsAggregator implements SizedBucketAg
 
         bucketOrds = LongKeyedBucketOrds.build(context.bigArrays(), cardinality);
 
-        fastFilterContext = new FastFilterRewriteHelper.FastFilterContext(
-            context,
-            new DateHistogramAggregationType(
-                valuesSourceConfig.fieldType(),
-                valuesSourceConfig.missing() != null,
-                valuesSourceConfig.script() != null,
-                hardBounds
-            )
-        );
-        if (fastFilterContext.isRewriteable(parent, subAggregators.length)) {
-            fastFilterContext.buildRanges(Objects.requireNonNull(valuesSourceConfig.fieldType()));
-        }
-    }
+        optimizationContext = new OptimizationContext(new DateHistogramAggregatorBridge() {
+            @Override
+            protected boolean canOptimize() {
+                return canOptimize(valuesSourceConfig);
+            }
```

Comment:
You can define specific constructors in bridge class to avoid inline implementations.

Reply:
Not sure how to do this. This canOptimize method is called from OptimizationContext to invoke any logic related to check canOptimize.

Reply:
Specifically don't want to pass in to the bridge class and make it a instance level variable.
Only takes in the necessary variables through calling method, so its scoped is restricted to the method